airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/SourceDefinitionsHandler.java

```diff
@@ -233,6 +238,20 @@ public SourceDefinitionRead getSourceDefinitionForWorkspace(final SourceDefiniti
   }
 
   public SourceDefinitionRead createCustomSourceDefinition(final CustomSourceDefinitionCreate customSourceDefinitionCreate) throws IOException {
+    final UUID idempotencyKey = customSourceDefinitionCreate.getIdempotencyKey();
+    if (idempotencyKey != null) {
+      Optional<StandardSourceDefinition> found = sourceService.getStandardSourceDefinitionByIdempotencyKey(idempotencyKey);
+      if (found.isPresent()) {
+        final StandardSourceDefinition sourceDefinition = found.get();
+        final ActorDefinitionVersion actorDefinitionVersion;
+        try {
+          actorDefinitionVersion = configRepository.getActorDefinitionVersion(sourceDefinition.getDefaultVersionId());
+          return buildSourceDefinitionRead(sourceDefinition, actorDefinitionVersion);
+        } catch (ConfigNotFoundException e) {
+          // not found, so must create
+        }
```

Comment:
I'm not too happy with this. I suspect this is not a situation that you can get in, since the existence of the sourcedef implies it has at least one version (i would think). But i have to catch this due to `buildSourceDefinitionRead` throwing it.